src/Android/Avalonia.Android/Platform/AndroidNativeControlHostImpl.cs

```diff
@@ -0,0 +1,187 @@
+#pragma warning restore IDE0017 // Simplify object initialization
+                attachment.AttachedTo = this;
+                return attachment;
+            }
+            catch
+            {
+                attachment?.Dispose();
+                holder?.Destroy();
+                throw;
+            }
+        }
+
+        public INativeControlHostControlTopLevelAttachment CreateNewAttachment(IPlatformHandle handle)
+        {
+            return new AndroidNativeControlAttachment(handle)
+            {
+                AttachedTo = this
+            };
+        }
+
+        public bool IsCompatibleWith(IPlatformHandle handle) => handle.HandleDescriptor == AndroidViewControlHandle.AndroidDescriptor;
+
+        class AndroidNativeControlAttachment : INativeControlHostControlTopLevelAttachment
+        {
+            // ReSharper disable once NotAccessedField.Local (keep GC reference)
+            private IPlatformHandle? _child;
+            private View? _view;
+            private AndroidNativeControlHostImpl? _attachedTo;
+
+            public AndroidNativeControlAttachment(IPlatformHandle child)
+            {
+                _child = child;
+
+                _view = (child as AndroidViewControlHandle)?.View
+                    ?? Java.Lang.Object.GetObject<View>(child.Handle, global::Android.Runtime.JniHandleOwnership.DoNotTransfer);
+            }
+
+            [MemberNotNull(nameof(_view))]
+            private void CheckDisposed()
+            {
+                if (_view == null)
+                    throw new ObjectDisposedException(nameof(AndroidNativeControlAttachment));
+            }
+
+            public void Dispose()
+            {
+                if (_view != null && _attachedTo?._avaloniaView is ViewGroup parent)
+                {
+                    parent.RemoveView(_view);
+                }
+                _child = null;
+                _attachedTo = null;
+                _view?.Dispose();
+                _view = null;
+            }
+
+            public INativeControlHostImpl? AttachedTo
+            {
+                get => _attachedTo;
+                set
+                {
+                    CheckDisposed();
+
+                    var oldAttachedTo = _attachedTo;
+                    _attachedTo = (AndroidNativeControlHostImpl?)value;
+                    if (_attachedTo == null)
+                    {
+                        oldAttachedTo?._avaloniaView.RemoveView(_view);
+                    }
+                    else
+                    {
+                        _attachedTo._avaloniaView.AddView(_view);
+                    }
+                }
+            }
+
+            public bool IsCompatibleWith(INativeControlHostImpl host) => host is AndroidNativeControlHostImpl;
+
+            public void HideWithSize(Size size)
+            {
+                CheckDisposed();
+                _view.Visibility = ViewStates.Gone;
```

Comment:
We've had complaints about "hidden view being of invalid size" on macOS. That's why method is called "HideWithSize". Not sure if it's applicable to Android.

Reply:
I can imagine when somebody would want to get bounds of invisible element, but why only size then?

Reply:
Anyway, I will add it